gslib/utils/copy_helper.py

```diff
@@ -1976,8 +2010,9 @@ def _UploadFileToObject(
       logger, allow_splitting, upload_url, dst_url, src_obj_size,
       gsutil_api, canned_acl=global_copy_helper_opts.canned_acl,
       kms_keyname=dst_obj_metadata.kmsKeyName)
-  non_resumable_upload = (upload_size < ResumableThreshold() or
-                          src_url.IsStream() or src_url.IsFifo())
+  non_resumable_upload = (
+    (0 if upload_size is None else upload_size) < ResumableThreshold() or
```

Comment:
This first parenthetical can be simplified to `upload_size or 0` -- even if it's not None but still falsy (meaning it's zero), we'll still end up with 0.